### DISCUSSION TITLE
fail to run charm with T1 and T2

### DISCUSSION THREAD

**Discussion by romainVala** [simnibs/simnibs#227] [7.Creating head models] [answered]
Hello
I use charm version 4.0.1, and it works fine with a single T1, but when I try to add a T2 in the command I get an error

here is the command I run
`charm --forceqform suj20 v_0020_S2_T1w_MPR.nii v_0020_S3_T2W_SPC.nii`

and I get the errror 
```
RuntimeError: /__w/charm-gems/charm-gems/ITK-install/include/ITK-4.13/itkImageFileReader.hxx:143:
 Could not create IO object for reading file /network/xxx ..(curent path) xxx /m2m_suj20/T2_reg.nii.gz
The file doesn't exist. 
Filename = /network/lustre/xxx ..(curent path) xxx/m2m_suj20/T2_reg.nii.gz

many thanks for your help

Romain


```

**Comment by oulap** [comment]
The charm command expects a subject name, then the scans and finally the flags, so try calling it like this:

`charm  suj20 v_0020_S2_T1w_MPR.nii v_0020_S3_T2W_SPC.nii --forceqform`

**Reply by romainVala** [reply]
sorry, but I get the exact same error

**Reply by oulap** [reply]
Where are you running this? The filepath looks a little strange:

`
/network/lustre/xxx ..(curent path) xxx/m2m_suj20/T2_reg.nii.gz`

**Reply by romainVala** [reply]
Sorry I just remove the true path for privacy, but yes this is a simple linux path. And again the same command works fine if I only provide the T1

I finally succeed to debug via ipython, and I made a change : 
in charm_main.py line 665, I remove the indentation, so that the save is perform even if do not need to do a squeeze so the code is now

```
                T2_tmp = nib.load(T2)
                T2_tmp = _check_q_and_s_form(T2_tmp, force_qform, force_sform)

                if (np.array(T2_tmp.shape) == 1).any():
                    data_tmp = np.squeeze(T2_tmp.get_fdata())
                    T2_tmp = nib.Nifti1Image(data_tmp, T2_tmp.affine)

                nib.save(T2_tmp, T2_reg) #remove the indent to alway do the save
                charm_utils._registerT1T2(T1, T2_reg, T2_reg)

```

Actually I do not really understand why I need to add the --forceqform flag, since in my case both qform and sform are identical (may there are some rounding error when you compare the 2, which may happen with non axial acquisition). but it is not a big deal so I add it, (note that the flag position in the command does not matter)

**Reply by oulap** [reply, marked as answer]
Aah, I see. I thought I fixed that bug in this commit:

https://github.com/simnibs/simnibs/commit/e7629b0ce60cf55b796f091875de292f99edd6b2

which should have made it's way to v4.0.1. I'll double-check that. 

The q/sform can sometimes be a mess, especially if the simulations are later used as input to actual stimulation, so we decided to make it clear which form is used even when there are only  small differences in between them.
